tf_quant_finance/black_scholes/approximations/american_option.py

```diff
@@ -0,0 +1,290 @@
+    raise ValueError('At most one of continuous_dividends and cost_of_carries '
+                     'may be supplied')
+
+  # For ease, not split out - cost of calculating eu call price for all small in
+  # comparison to american prices
+  eu_prices = vanilla_prices(
+    volatilities=volatilities, strikes=strikes, expiries=expiries, spots=spots,
+    forwards=forwards, discount_rates=discount_rates,
+    continuous_dividends=continuous_dividends, cost_of_carries=cost_of_carries,
+    dtype=dtype, name=name)
+
+  # Split out american
+  cost_of_carries = discount_rates - continuous_dividends \
+    if cost_of_carries is None else cost_of_carries
+
+  where_call = tf.constant(
+    True) if is_call_options is None else tf.reduce_any(is_call_options)
+  calculate_eu = tf.logical_and(where_call, cost_of_carries >= discount_rates)
+
+  if forwards is not None:
+    expiries = tf.convert_to_tensor(expiries, dtype=dtype, name="expiries")
+    spots = tf.convert_to_tensor(
+      forwards * tf.exp(-cost_of_carries * expiries), dtype=dtype, name="spots")
+  else:
+    expiries = tf.convert_to_tensor(expiries, dtype=dtype, name="expiries")
+    spots = tf.convert_to_tensor(spots, dtype=dtype, name="spots")
+
+  if is_call_options is not None:
+    is_call_options = tf.convert_to_tensor(is_call_options, dtype=tf.bool)
+  strikes = tf.convert_to_tensor(strikes, dtype=dtype, name="strikes")
+  volatilities = tf.convert_to_tensor(
+    volatilities, dtype=dtype, name="volatilities")
+  if continuous_dividends is not None:
+    cost_of_carries = tf.convert_to_tensor(
+      discount_rates - continuous_dividends, dtype=dtype,
+      name="cost_of_carries")
+  else:
+    cost_of_carries = tf.convert_to_tensor(
+    cost_of_carries, dtype=dtype, name="cost_of_carries")
+
+  discount_rates = tf.convert_to_tensor(
+    discount_rates, dtype=dtype, name="discount_rates")
+
+  am_prices, converged, failed = _option_price(
+    sigma=volatilities, x=strikes, t=expiries, s=spots, r=discount_rates,
+    b=cost_of_carries, is_call_options=is_call_options, dtype=dtype,
+    max_iterations=max_iterations, tolerance=tolerance, name=name)
+
+  return tf.where(calculate_eu, eu_prices, am_prices), converged, failed
+
+
+def _option_price(*,
+                  sigma,
+                  x,
+                  t,
+                  r,
+                  b,
+                  s=None,
+                  is_call_options=None,
+                  dtype=None,
+                  max_iterations=20,
+                  tolerance=1e-8,
+                  name=None):
+  """Computes prices of American options using the Baron-Adesi Whaley approximation."""
+
+    # The naming convention will align variables with the variables named in
+    # reference [1], but made lower case, and differentiating between put and
+    # call option values with the suffix _put and _call.
+    # [1] https://deriscope.com/docs/Barone_Adesi_Whaley_1987.pdf
```

Comment:
I guess it's failing because of extra spaces here?

Reply:
took those away. What was failing? Is there some kind of linter I could run to make the code consistent style? (at work we use black and it makes this a lot easier :) ) (but I do note that the style of tfqf is not the same as black so I can't user that) Do you maybe have a pylint file that could at least highlight my mistakes?